rfc/organizing-work.md

```diff
@@ -0,0 +1,58 @@
+
+![status:draft](https://img.shields.io/badge/status-draft-yellow.svg?style=flat-square)
+
+## Editors
+
+- [Alan Shaw](https://github.com/alanshaw), [Protocol Labs](https://protocol.ai/)
+
+## Authors
+
+- [Alan Shaw](https://github.com/alanshaw), [Protocol Labs](https://protocol.ai/)
+
+## Abstract
+
+Formalize a process for working that ensures a consistent and up to date view of the work items in our pipeline. Serves to document a process that can be onboarded and observed by others.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+## Out of scope
+
+* The tools or locations used to document work items. i.e. Github issues, Jira, Notion etc.
+* Roadmapping/general high level direction setting.
+
+## Pipeline
+
+Generally a work item would move from inception to completion through a pipeline in the following order:
+
+1. **Heap** - anyone can pile stuff onto the heap. It's just a place to track things that we may want to work on in the future. It can be of arbitrary size and detail.
+
+    Items that are of ambiguous _certainty_ SHOULD be accompanied by an [RFC](https://github.com/web3-storage/rfc) - a PR that allows others to comment in an open and structured way on the item. For example, an RFC may be a new idea or a way of achieving a goal that could be done in multiple ways. An RFC allows others to validate the idea, question or assumption to determine with more certainty if it should be worked on.
+    
+    All members SHOULD read RFC documents and provide feedback to allow the author to determine validity of the work item and adjust or clean up if necessary.
+
+2. **Stack** - prioritized work items. _Unlike_ a conventional stack, items are added at a position that denotes their relative priority, however, as per a conventional stack, items are usually "popped" from the top.
+
+    Items SHOULD be moved from the heap to the stack as part of a regular lightweight planning exercise that SHOULD include members of the group that will be working on the items. At this point a short description that details the _product benefits_ and _desired end state/outcome(s)_ of the system MUST be added if not already present.
+
+    Items on the stack SHOULD be prioritized according to the general roadmap as well as other factors, such as, but limited to, how on fire things are. Items in the stack may be re-prioritized during group planning.
+    
+    Multiple items in the heap MAY result in a _single_ item on the stack.
+
+    Larger items whose outputs are an implementation of some kind MUST also result in a PR to the [specs](https://github.com/web3-storage/specs/) repo.
+
+3. **Phase** - a short period of time where items on the stack are worked on. It focuses group effort on a particular theme and is analogous to a "sprint" in Scrum.
```

Comment:
This is the one I have the most trouble with. I think we're trying to combine two seperate ideas into a single one.

The first is just a cycle for getting stuff done -- i.e. a sprint, though really I just mean the combination of backlog grooming, sprint planning, and reflection. A "rhythm" to work if you will. I think at least till we are larger, there should just be one of these.

The second is grouping work -- how we insure people can take on big themes together and not get silo'd alone or stuck doing a single thing. I really like the idea of temporary sub team. Essentially, we're all on web3.storage but we can form groupings that work closely for a period of time on a big deliverable -- that can span a month, or even, in some cases longer.

My anxiety about trying to combine both is that we end up making "what's happening" extremely difficult, and impose artificial time contraints on work that may take much longer than a sprint to do.

Am I missing something here in the explanation? Am I not getting what a phase is supposed to do?

Reply:
I think this is a really important thing to tease apart.